### ISSUE TITLE
OSError: Could not load shared object file: libllvmlite.so

### ISSUE THREAD

**Issue by Namburger** [librosa/librosa#1116]
### Description
I was trying to install libora on a aarch64 machine by following the docs:
```
$ sudo apt install python3-dev
$ pip3 install Cython --user
$ wget https://github.com/librosa/librosa/archive/0.7.2.tar.gz
$ tar -xvf 0.7.2.tar.gz librosa-0.7.2/
$ cd librosa-0.7.2/
$ python3 setup.py install
```
Ran into outdated TBB error:
```
numba/np/ufunc/tbbpool.cpp:89:108: error: no matching function for call to ‘tbb::internal::task_scheduler_observer_v3::task_scheduler_observer_v3(tbb::interface7::task_arena&)’
fix_tls_observer(tbb::task_arena &arena, int mask) : tbb::task_scheduler_observer(arena), mask_val(mask)
```
So I began building TBB from source (also had to build cmake because linux amirite):
```
$ sudo apt install libffi-dev
$ git clone https://github.com/wjakob/tbb.git
$ mkdir build && cd build
$ cmake ..
$ make
$ sudo make install
```
Back to librosa, finally got everything to compile:
```
$ sudo python3 setup.py install
```
but when I try to use the library, it said that I'm missing a share object from `libllvmlite.so`:
```
python3 -c 'print(__import__("librosa").__path__)'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.7/dist-packages/librosa-0.7.2-py3.7.egg/librosa/__init__.py", line 12, in <module>
    from . import core
  File "/usr/local/lib/python3.7/dist-packages/librosa-0.7.2-py3.7.egg/librosa/core/__init__.py", line 125, in <module>
    from .time_frequency import *  # pylint: disable=wildcard-import
  File "/usr/local/lib/python3.7/dist-packages/librosa-0.7.2-py3.7.egg/librosa/core/time_frequency.py", line 11, in <module>
    from ..util.exceptions import ParameterError
  File "/usr/local/lib/python3.7/dist-packages/librosa-0.7.2-py3.7.egg/librosa/util/__init__.py", line 77, in <module>
    from .utils import *  # pylint: disable=wildcard-import
  File "/usr/local/lib/python3.7/dist-packages/librosa-0.7.2-py3.7.egg/librosa/util/utils.py", line 10, in <module>
    import numba
  File "/usr/local/lib/python3.7/dist-packages/numba-0.49.1-py3.7-linux-aarch64.egg/numba/__init__.py", line 15, in <module>
    from numba.core import config
  File "/usr/local/lib/python3.7/dist-packages/numba-0.49.1-py3.7-linux-aarch64.egg/numba/core/config.py", line 16, in <module>
    import llvmlite.binding as ll
  File "/usr/local/lib/python3.7/dist-packages/llvmlite-0.32.1-py3.7.egg/llvmlite/binding/__init__.py", line 4, in <module>
    from .dylib import *
  File "/usr/local/lib/python3.7/dist-packages/llvmlite-0.32.1-py3.7.egg/llvmlite/binding/dylib.py", line 3, in <module>
    from llvmlite.binding import ffi
  File "/usr/local/lib/python3.7/dist-packages/llvmlite-0.32.1-py3.7.egg/llvmlite/binding/ffi.py", line 153, in <module>
    raise OSError("Could not load shared object file: {}".format(_lib_name))
OSError: Could not load shared object file: libllvmlite.so
```
**llvmlite** seems to be installed:
```
$ python3 -c 'print(__import__("llvmlite").__path__)'
['/usr/local/lib/python3.7/dist-packages/llvmlite-0.32.1-py3.7.egg/llvmlite']
```
Although I can't find any .so file, I would really appreciate the help if you can point me to the right direction.

#### Versions
```
import platform; print(platform.platform())
Linux-4.14.98-imx-aarch64-with-Mendel-mendel-day-day
import sys; print("Python", sys.version)
Python 3.7.3
import numpy; print("NumPy", numpy.__version__)
NumPy 1.16.2
import scipy; print("SciPy", scipy.__version__)
SciPy 1.1.0
import librosa; print("librosa", librosa.__version__)
...
    raise OSError("Could not load shared object file: {}".format(_lib_name))
OSError: Could not load shared object file: libllvmlite.so
```
Thanks

**Comment by bmcfee** [comment]
This looks like an llvmlite packaging / build issue to me.

I don't know for sure, but my guess is that the llvmlite package doesn't have wheels for your platform.  So the python package exists to bind to the library, but the library itself isn't there (as you report).  I gather that llvmlite was installed via pip, but maybe it's worth trying to install from source?  Failing that, I'd suggest to git up the [llvmlite github](https://github.com/numba/llvmlite).

**Comment by Namburger** [comment]
@bmcfee Thanks, I also tried installing llvmlite via pip and that process went quite smoothly, I'm still confused as to why I couldn't tracing any .so file but I'll try compiling that form source tomorrow :)
Maybe I'll give you guys update so that other can reference, will close this out for now

**Comment by Namburger** [comment]
Unfortunately, failed to build llvmlite because of missing `llvm-config` static lib haha

**Comment by stuartarchibald** [comment]
@Namburger I think you need to do this: https://github.com/numba/llvmlite/issues/588#issuecomment-626800619 `llvmlite` statically links to LLVM and your system is not providing the static libraries.

**Comment by Namburger** [comment]
@stuartarchibald @bmcfee 
Yelp, it works, need more works on my googling abilities :p

**Comment by Namburger** [comment]
FYI, for those who are interested in getting this working on the [Coral Dev Board](https://coral.ai/products/dev-board/):
https://gist.github.com/Namburger/82e86f2d3514288aa46c9183fc53c809

**Comment by inderpreetsingh01** [comment]
Hi @Namburger , I faced the similar issue and followed the procedure you have mentioned, but it gave me this error

E: Unable to locate package libllvm7:arm64
E: Unable to locate package llvm-7
E: Unable to locate package llvm-7-dev
E: Unable to locate package llvm-7-runtime
E: Unable to locate package python3-llvmlite

**Comment by Namburger** [comment]
@inderpreetsingh01 This was made for aarch64 platform, and also dependent of OS, I guess it is more of a guide than to follow it specifically. What OS are you using?

**Comment by inderpreetsingh01** [comment]
@Namburger  I am using ubuntu 16.04 and aarch64 platform only, specifically jetson.

**Comment by Namburger** [comment]
I see, sorry, I'm not sure as I've not used the jetson nano :(

**Comment by cdeln** [comment]
I bumped into this issue on debian 10, so I will leave some breadcrumbs for other poor souls who bump into this in the future. I traced down the issue to that libllvmlite.so requires  GLIBCXX_3.4.26 , but debian 10 comes shipped with gcc 8 and GLIBCXX version up to 3.4.25 (run `strings /usr/lib/x86_64-linux-gnu/libstdc++.so.6 | grep GLIBC` to see versions available) . I followed some upgrade steps (the accepted answer [in this thread](https://stackoverflow.com/questions/61590804/how-to-install-gcc-9-on-debianlatest-docker) ) and then it worked.

**Comment by jtremblay** [comment]
I just want to add my breadcrumbs too as I spent way too much time trying to solve this issue. Long story short, I needed to install llvm-lite on a ubuntu18.04 OS using python 3.6.5. What ultimately worked was to download the binaries for ubuntu 16.04:
https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.1/clang+llvm-10.0.1-x86_64-linux-gnu-ubuntu-16.04.tar.xz, untar the whole thing and then, export LLVM_CONFIG=/path/to/bin/llvm-config
Then download the python llvm-lite 0.36.0 and install it manually.
